### PR TITLE
Add missing bindings and documentation for MultiplayerPeerExtension

### LINKED ISSUE #74628: MultiplayerPeerExtension requires more overrides than docs specify
### Godot version

4.0-stable

### System information

Linux

### Issue description

https://docs.godotengine.org/en/stable/classes/class_multiplayerpeerextension.html#class-multiplayerpeerextension says:

> All the methods below must be implemented to have a working custom multiplayer implementation

I've implemented all methods listed in the docs, but I get:

```
ERROR: Required virtual method MultiplayerPeerExtension::_get_packet_channel must be overridden before calling.
   at: _gdvirtual__get_packet_channel_call (scene/main/multiplayer_peer.h:142)
ERROR: Required virtual method MultiplayerPeerExtension::_get_packet_mode must be overridden before calling.
```

These aren't mentioned in the docs. I'm not sure if this is just a doc issue or an implementation issue.

### Steps to reproduce

1. Extend `MultiplayerPeerExtension` in GDScript
2. Implement all methods listed in the docs
3. Try to use it as a multiplayer peer.

### Minimal reproduction project

[example.zip](https://github.com/godotengine/godot/files/10926357/example.zip)

### PR DESCRIPTION
**Pull request by dsnopek**
Fixes #74628

### CODE REVIEW

File: scene/main/multiplayer_peer.cpp [hunk 1]
```diff
@@ -201,6 +201,9 @@ void MultiplayerPeerExtension::_bind_methods() {
 	GDVIRTUAL_BIND(_get_packet_script)
 	GDVIRTUAL_BIND(_put_packet_script, "p_buffer");
 
+	GDVIRTUAL_BIND(_get_packet_channel);
+	GDVIRTUAL_BIND(_get_packet_mode);
```
Comment (Faless) [inline scene/main/multiplayer_peer.cpp:205]: I think we are also missing:

`GDVIRTUAL_BIND(_is_server_relay_supported);`

which I forgot to add too.

Comment (dsnopek) [reply scene/main/multiplayer_peer.cpp:205]: Ah, yes, thanks! I've added it in my latest push

File: doc/classes/MultiplayerPeerExtension.xml [hunk 2]
```diff
@@ -77,6 +89,7 @@
 			<return type="int" />
 			<description>
 				Called when the unique ID of this [MultiplayerPeer] is requested (see [method MultiplayerPeer.get_unique_id]).
+				It must be between 1 and 2147483647.
```
Comment (YuriSizov) [inline doc/classes/MultiplayerPeerExtension.xml:92]:
```suggestion
				It must be between [code]1[/code] and [code]2147483647[/code].
```

Also doesn't need to be a new paragraph.

**Review by YuriSizov — APPROVED** [review]
LGTM, one nitpick though.

**Comment by YuriSizov** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 4.0.3.

**Pull request merged**